linkpreview.go

```diff
@@ -1,17 +1,38 @@
 package linkpreview
 
 import (
+	"errors"
 	"io"
 	"net/http"
+	"net/url"
+	"strings"
 
 	"github.com/PuerkitoBio/goquery"
 )
 
+func validateURL(link string) error {
+	parsedURL, err := url.ParseRequestURI(link)
+	if err != nil || !strings.HasPrefix(parsedURL.Scheme, "http") {
+		return errors.New("invalid URL")
+	}
+	return nil
+}
+
 func Parse(link string, options ...ConfigOption) (ParserResult, error) {
+	if err := validateURL(link); err != nil {
```

Comment:
Maybe this is not necessary as `http.NewRequest()` and `http.DefaultClient.Do()` should handle the invalid URL correctly.